src/main/resources/templates/macros/report/element.vm

```diff
@@ -9,12 +9,13 @@
 
   #includeTags($element.getTags())
 
-  <span data-toggle="collapse" class="#if ($element.getElementStatus().isPassed()) collapsed #end collapsable-control" data-target="#element-$element.hashCode()">
+  #set($elementId = $counter.getNext())
+  <span data-toggle="collapse" class="#if ($element.getElementStatus().isPassed()) collapsed #end collapsable-control" data-target="#element-$elementId">
     #includeBrief($element.getKeyword(), $element.getElementStatus(), $element.getName(), true)
   </span>
   <div class="description indention">$element.getDescription()</div>
 
-  <div id="element-$element.hashCode()" class="collapse collapsable-details #if(!$element.getElementStatus().isPassed()) in #end">
+  <div id="element-$elementId" class="collapse collapsable-details #if(!$element.getElementStatus().isPassed()) in #end">
```

Comment:
I think hardcoding a prefix into the counter isn't ideal. It will work, but can confuse when reading the code. I would prefer to keep this as-is. Another option is to have an API to specify the prefix so each template can set it. But since counter is a global thing (at least for a single page), it can easily 'leak' across templates and cause further confusion when reading the generated code (but it will still work).